man/systemd.kill.xml

```diff
@@ -135,9 +136,23 @@
       <varlistentry>
         <term><varname>SendSIGKILL=</varname></term>
         <listitem><para>Specifies whether to send
-        <constant>SIGKILL</constant> to remaining processes after a
-        timeout, if the normal shutdown procedure left processes of
-        the service around. Takes a boolean value. Defaults to "yes".
+        <constant>SIGKILL</constant> (or the signal specified by
+        <varname>FinalKillSignal=</varname>) to remaining processes
+        after a timeout, if the normal shutdown procedure left
+        processes of the service around. Takes a boolean value.
+        Defaults to "yes".
+        </para></listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><varname>FinalKillSignal=</varname></term>
+        <listitem><para>Specifies which signal to send to remaining
+        processes after a timeout if <varname>SendSIGKILL=</varname>
+        is set to"yes". Developers can find it useful to set this to
+        <constant>SIGQUIT</constant> to generate a coredump to
```

Comment:
Hmm, for watchdog events we use SIGABRT, I think it would make sense to at least mention that here too as suitable.

Also, I think this should carry some additional wording clarifying that SIGTERM is not suitable here, i.e. signals that are typically caught and processed by the services themselves.

Reply:
Both `FinalKillSignal=SIGQUIT` and `FinalKillSignal=SIGABRT` work to produce a coredump, so I can mention both, however...

I maybe wrong, but I believe the semantic difference between the two is:

- SIGQUIT is intended to produce a coredump from a signal originating external to the process being signaled
- SIGABRT is intented for a process to produce a coredump of itself (via a call to `abort()` or `raise(SIGABRT)` or `kill(getpid(), SIGSEGV)`)

Reply:
well, if you look at the Linux man page, neither fits perfectly. About SIGQUIT it says: "Quit from keyboard". About SIGABRT it says: "Abort signal from abort(3)". 

If you look at POSIX, it says about SIGQUIT: "Terminal quit signal." About SIGABRT it says: "Process abort signal."

Now, we are not a keyboard nor a terminal, so SIGQUIT doesn't fully apply. We are also not abort(3), hence SIGABRT doesn't fully apply either. (POSIX is nicely vague however, which I figure could git us well)

Now, what I find the most relevant though is that we do use SIGABRT for the watchdog case, and it's kinda nice to stick to the same one here for other timeout related stuff. And the final killing really is about a timeout, i.e. when a process got SIGTERM but doesn't react to it in time...

Reply:
I was inferring the semantic difference based on the fact that a signal originating from a keyboard/terminal is external to the process and that a signal originating from an `abort()` call originates internally from the process itself. But, both produce the desired end result for my use case :)